Firmware/ultralcd.cpp

```diff
@@ -5121,6 +5121,16 @@ static void lcd_sheet_menu()
     MENU_END();
 }
 
+//! @brief Set printer state
+//! Sets the printer state for next print via LCD menu
+//! @endcode
+static void lcd_printer_status_toggle()
+{
+    if (printer_status == PrinterStatus::NotReady) printer_status = PrinterStatus::IsReady;
+    else printer_status = PrinterStatus::NotReady;
+    enquecommandf_P(PSTR("M118 A1 action:%s"), (printer_status == PrinterStatus::NotReady) ? "not_ready" : "ready");
```

Comment:
Optimisation idea:

Please add to `messages.cpp/h`:

```c
const char MSG_OCTOPRINT_NOT_READY[] PROGMEM_N1 = "// action:not_ready"; ////
const char MSG_OCTOPRINT_READY[] PROGMEM_N1 = "// action:ready"; ////
```

And change:
`enquecommandf_P(PSTR("M118 A1 action:%s"), (printer_status == PrinterStatus::NotReady) ? "not_ready" : "ready");` 
into:
`enquecommandf_P(PSTR("M118 %S"), (printer_status == PrinterStatus::NotReady) ? MSG_OCTOPRINT_NOT_READY : MSG_OCTOPRINT_READY);`

Change in memory:
Flash: +16 bytes
SRAM: -10 bytes

Note that `"not_ready" : "ready"` without `PSTR()` will use up SRAM instead of Flash.